api/auth.go

```diff
@@ -0,0 +1,155 @@
+package api
+
+import (
+	"context"
+	"fmt"
+	"github.com/golang-jwt/jwt/v4"
+	"golang.org/x/crypto/bcrypt"
+	"gym_management_system/db"
+	"gym_management_system/errors"
+	"log"
+	"net/http"
+	"os"
+	"time"
+)
+
+func validPassword(reqPw, encPw string) bool {
+	return bcrypt.CompareHashAndPassword([]byte(encPw), []byte(reqPw)) == nil
+}
+
+func HashPassword(password string) (string, error) {
+	pw, err := bcrypt.GenerateFromPassword([]byte(password), bcrypt.DefaultCost)
+	return string(pw), err
+}
+
+func createJWT(account *db.Account, exp time.Time) (string, error) {
+	claims := &Claims{
+		Id: account.Id,
+		RegisteredClaims: jwt.RegisteredClaims{
+			ExpiresAt: jwt.NewNumericDate(exp),
+		},
+	}
+	token := jwt.NewWithClaims(jwt.SigningMethodHS256, claims)
+	secret := os.Getenv("JWT_SECRET")
+	return token.SignedString([]byte(secret))
+}
+
+func validateJWT(tokenString string, claims jwt.Claims) (*jwt.Token, error) {
+	secret := os.Getenv("JWT_SECRET")
```

Comment:
Or that someone changes the variable when the program is running. Ideally, you would want to have it parsed once in the `main.go` when initializing the app and pass it around in some struct.